src/borg/archiver.py

```diff
@@ -1236,10 +1246,15 @@ def _delete_repository(self, args, repository):
                 msg.append("This repository seems to have no manifest, so we can't tell anything about its "
                            "contents.")
             else:
-                msg.append("You requested to completely DELETE the repository *including* all archives it "
+                if self.output_list:
+                    msg.append("You requested to completely DELETE the repository *including* all archives it "
                            "contains:")
-                for archive_info in manifest.archives.list(sort_by=['ts']):
-                    msg.append(format_archive(archive_info))
+                    for archive_info in manifest.archives.list(sort_by=['ts']):
+                        msg.append(format_archive(archive_info))
+                else:
+                    msg.append("You requested to completely DELETE the"
+                               "repository *including* %d archives it"
+                               "contains." % len(manifest.archives))
```

Comment:
nitpick: we do not obey 80 chars/line, but (iirc) 120 (see flake8 configuration in setup.cfg or so).

so, you could put this into 1 or 2 lines, not 3.